plugins/lsp/language-servers/_mk-lsp.nix

```diff
@@ -78,6 +79,25 @@ in
             cmd cfg
           else
             cmd;
+        defaultText = lib.literalMD ''
+          null when `package` is null, otherwise ${
+            if args ? cmdText || builtins.isFunction cmd then
+              let
+                literal = lib.options.renderOptionValue cmdText;
+                inherit (literal) text;
+              in
+              if literal._type == "literalMD" then
+                text
+              else if lib.hasInfix "\n" text || lib.hasInfix "``" text then
+                "\n\n```\n${text}\n```"
+              else
+                "`` ${text} ``"
+            else if cmd == null then
+              "null"
+            else
+              "`[ ${lib.concatMapStringsSep " " builtins.toJSON cmd} ]`"
+          }
+        '';
```

Comment:
Internally, I believe we only ever supply `cmd` as a list. So only the last two branches here (`if null` and `else`) will ever be used.

It'd be a breaking change for anyone using mkLsp out-of-tree, but for simplicity we could drop support for `cmd` being a function?

Reply:
I'm not sure if we expose in an easily accessible manner `mkLsp`, so I don't think it's a big breaking change

Reply:
> Also, is it really necessary to change `cmd`'s default to null when `package` is null? Why not just set `cmd`'s default unconditionally?

The implemention is quite old, maybe before I started to contribute. The direction we took recently was to try and make most things less magic, so I'd be in favour of less complex behaviour